app/controllers/admin/settings_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Admin
+  class SettingsController < ApplicationController
+    def index; end
+
+    def create
```

Comment:
That's because RESTful route for PATCH requires an `:id`. There is no such thing as a *single* resource in rails: https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default

The documentation of `rails-cached-settings` is using `create` for that reason: https://github.com/huacnlee/rails-settings-cached#how-to-manage-settings-in-the-admin-interface

Reply:
Okay, makes sense. You can still just define routes explicitly, you don’t have to follow those convention if you feel they aren’t a good fit for the use case.

```
get '/settings', to: 'settings#index'
post '/settings', to: 'settings#update'

Reply:
..and then also Rails form helpers don't work out-of-the-box :disappointed: I tried exactly that before :confused:

Reply:
I just tried this out locally and it worked without any problems (at least I was able to update settings and didn’t notice any difference.)

I used the route definitions I mentioned above and changed the form action to `settings_path` (instead of `admin_settings_path`).

Reply:
I just tried this out locally and it worked without any problems (at least I was able to update settings and didn’t notice any difference.)

I used the route definitions I mentioned above and changed the form action to `settings_path` (instead of `admin_settings_path`).